src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -3308,6 +3385,50 @@ GetRunnableBackgroundTask(void)
 }
 
 
+/*
+ * GetTaskTokens
+ * Checks whether the node ids in the task tokens are available
+ * If at least one of them is not available, it returns false.
+ * If yes, it takes all the needed tokens and returns true.
+ */
+static bool
+GetTaskTokens(HTAB *ParallelMovesPerNode, BackgroundTask *task)
+{
+	if (task->nodeTokens)
+	{
+		int node_token;
+
+		/* first check if all tokens are available*/
+		foreach_int(node_token, task->nodeTokens)
+		{
+			bool found;
+			ParallelMovesPerNodeEntry *hashEntry = hash_search(
+				ParallelMovesPerNode, &(node_token), HASH_ENTER, &found);
+			if (!found)
+			{
+				hashEntry->counter = 0;
+			}
+			else if (hashEntry->counter == MaxParallelMovesPerNode)
```

Comment:
It's possible that MaxParellelMovesPerNode is decreased. We should still block the needed tokens then.

```suggestion
			else if (hashEntry->counter >= MaxParallelMovesPerNode)
```